### DISCUSSION TITLE
(Godot/GodotJolt source) Replacing Godot Physics with Jolt in-engine?

### DISCUSSION THREAD

**Discussion by catboy-catfish** [godot-jolt/godot-jolt#822] [Questions]
I want to replace Godot Physics’ code with Jolt’s code in-engine so I don’t have to use the GDExtension. Even with only the Windows .DLL files included the GDExtension folder is still pretty big. I want to do this for my fork of Godot. How hard would this be?

**Comment by mihe** [comment]
> How hard would this be?

It's doable, but tedious. There have been other attempts at porting this extension over to a module (see [this one](https://github.com/EIRTeam/godot-jolt) for example), but it involves plenty of hacks, and is prone to breaking whenever I do large changes.

I'm also not convinced you'll save all that much in terms of binary size, or that it's worth the disproportionate effort in the first place. In the end you'll likely only ship with one of those DLLs per platform, which is ~3.5 MB of extra space. Cranking up the compression ratio on a couple of textures can probably make up for that pretty quickly. Even just compiling Godot itself with a different compiler/linker might net you bigger savings.

In any case, it's looking likely that this extension might get merged into Godot as an official module in a not too distant future, but I don't have a timeline for when that might be.
